libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -491,51 +508,22 @@ TEST_F(CppInterface, ErrorTagIsInputFileName) {
 }
 
 TEST_F(CppInterface, PreprocessingOnlyOption) {
-  options_.SetPreprocessingOnlyMode();
-  shaderc::CompilationResult result = compiler_.CompileGlslToSpv(
+  const PreprocessedSourceCompilationResult result = compiler_.PreprocessGlsl(
       kMinimalShaderWithMacro, shaderc_glsl_vertex_shader, "shader", options_);
   EXPECT_TRUE(CompilationResultIsSuccess(result));
-  EXPECT_THAT(result.GetData(), HasSubstr("void main(){ }"));
+  EXPECT_THAT(CompilerOutputAsString(result), HasSubstr("void main(){ }"));
 
   const std::string kMinimalShaderCloneOption =
       "#version 140\n"
       "#define E_CLONE_OPTION main\n"
       "void E_CLONE_OPTION(){}\n";
   CompileOptions cloned_options(options_);
-  shaderc::CompilationResult result_from_cloned_options =
-      compiler_.CompileGlslToSpv(kMinimalShaderCloneOption,
-                                 shaderc_glsl_vertex_shader, "shader",
-                                 cloned_options);
+  const PreprocessedSourceCompilationResult result_from_cloned_options =
+      compiler_.PreprocessGlsl(kMinimalShaderCloneOption,
+                               shaderc_glsl_vertex_shader, "shader",
+                               cloned_options);
   EXPECT_TRUE(CompilationResultIsSuccess(result_from_cloned_options));
-  EXPECT_THAT(result_from_cloned_options.GetData(),
-              HasSubstr("void main(){ }"));
-}
-
-TEST_F(CppInterface, PreprocessingOnlyModeFirstOverridesDisassemblyMode) {
-  // Sets preprocessing only mode first, then sets disassembly mode.
-  // Preprocessing only mode should override disassembly mode.
-  options_.SetPreprocessingOnlyMode();
-  options_.SetDisassemblyMode();
```

Comment:
Optional: I wonder if it's possible to create an equivalent test on `FileCompiler::SetDisassemblyFlag()`?